service/module.go

```diff
@@ -23,10 +23,19 @@ package service
 import (
 	"fmt"
 	"sync"
-
-	"github.com/uber-go/tally"
 )
 
+// ModuleProvider provides Modules.
+type ModuleProvider interface {
+	// Name is the name that will be used for a new Module
+	// if no name is given as a ModuleOption.
+	Name() string
+	// Create a new Module. The name of the Host and the scoping
+	// of associated functions on the Host will be done using a name
+	// provided by a ModuleOption, or by the Name on this ModuleProvider.
+	Create(Host) (Module, error)
```

Comment:
I'd rather just add this as a function to what is already there for easier implementation for others.